### PR TITLE
Make suggestion system string distance algorithm case-insensitive

### LINKED ISSUE #10546: [PowerShell 7 Preview 3] Typing commands with different casing returns different suggestions
This behavior exists on both, Windows and Linux.

For example:
Windows: Compare the suggestions for `get-wineven` and `Get-WinEven`
Linux: Compare the suggestions for `get-even` and `Get-Even`

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Fix #10546

## PR Context

Suggestion system uses a string distance algorithm which should be case-insensitive to find command names (and file names) which have chars in different cases.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by SteveL-MSFT — APPROVED** [review]

File: src/System.Management.Automation/utils/FuzzyMatch.cs [hunk 1]
```diff
@@ -23,14 +23,17 @@ public static bool IsFuzzyMatch(string string1, string string2)
...
+        /// Compute the case-insensitive distance between two strings.
         /// Based off https://www.csharpstar.com/csharp-string-distance-algorithm/.
         /// </summary>
         /// <param name="string1">The first string to compare.</param>
         /// <param name="string2">The second string to compare.</param>
         /// <returns>The distance value where the lower the value the shorter the distance between the two strings representing a closer match.</returns>
         public static int GetDamerauLevenshteinDistance(string string1, string string2)
         {
+            string1 = string1.ToUpperInvariant();
+            string2 = string2.ToUpperInvariant();
```
Comment (daxian-dbw) [inline src/System.Management.Automation/utils/FuzzyMatch.cs]: Just a question: Is it OK to use `Invariant` here? I'm not familiar with this code, but isn't it interactive and user-facing?

Comment (iSazonov) [reply src/System.Management.Automation/utils/FuzzyMatch.cs]: There is two kind of strings - command names and file names. My thoughts was that it is special identifiers and we could ignore culture. 
Although I do not have strict preferences.

**Comment by iSazonov** [comment]
@TravisEz13 I think we can merge.

**Comment by TravisEz13** [comment]
@daxian-dbw Please review again

**Comment by daxian-dbw** [comment]
My concern has been addressed.

**Pull request merged**